irodsManagerTest.py

```diff
@@ -398,5 +400,138 @@ def testTouch(self):
+        r = self.api.data_objects.stat('/tempZone/home/rods/anotherfile.txt')
+        self.assertEqual(r['irods_response']['status_code'], 0)
+        self.assertIn({
+            'name': 'jeb',
+            'zone': 'tempZone',
+            'type': 'rodsuser',
+            'perm': 'read_object'
+        }, r['permissions'])
+
+        # Remove the data objects
+        r = self.api.data_objects.remove('/tempZone/home/rods/anotherfile.txt')
+        self.assertEqual(r['irods_response']['status_code'], 0)
+
+        r = self.api.data_objects.remove('/tempZone/home/rods/newfile.txt')
+        self.assertEqual(r['irods_response']['status_code'], 0)
+
+        # Remove the resource
+        r = self.api.resources.remove('res')
+        self.assertEqual(r['irods_response']['status_code'], 0)
+
+
+    def testChecksums(self):
+        self.api.setToken(self.adminToken)
+
+        # Create a unixfilesystem resource.
+        r = self.api.resources.create('res', 'unixfilesystem', HOSTNAME, '/tmp/res', '')
+        # self.assertEqual(r['irods_response']['status_code'], 0)
+
+        # Create a non-empty data object
+        r = self.api.data_objects.write('/tempZone/home/rods/file.txt', 'These are the bytes being written to the object', 'res')
+        self.assertEqual(r['irods_response']['status_code'], 0)
+
+        # Replicate the data object
+        r = self.api.data_objects.replicate('/tempZone/home/rods/file.txt', dst_resource='res')
+        self.assertEqual(r['irods_response']['status_code'], 0)
+
+        # Show that there are two replicas
+        # TODO: Implement once query operations are completed
+
+        try:
+            # Calculate a checksum for the first replica
+            r = self.api.data_objects.calculate_checksum('/tempZone/home/rods/file.txt', replica_number=0)
+            self.assertEqual(r['irods_response']['status_code'], 0)
+
+            # Verify checksum information across all replicas.
+            r = self.api.data_objects.verify_checksum('/tempZone/home/rods/file.txt')
+            print(r)
+            self.assertEqual(r['irods_response']['status_code'], 0)
+        finally:
+            # Remove the data objects
+            r = self.api.data_objects.remove('/tempZone/home/rods/file.txt', 0, 1)
+            self.assertEqual(r['irods_response']['status_code'], 0)
+
+            # Remove the resource
+            r = self.api.resources.remove('res')
+            self.assertEqual(r['irods_response']['status_code'], 0)
+        
+    
+    def testTouch(self):
+        self.api.setToken(self.adminToken)
+
+        # Test touching non existant data object with no_create
+        r = self.api.data_objects.touch('/tempZone/home/rods/new.txt', 1)
+        self.assertEqual(r['irods_response']['status_code'], 0)
+
+        # Show that the object has not been created
+        r = self.api.data_objects.stat('/tempZone/home/rods/new.txt')
+        self.assertEqual(r['irods_response']['status_code'], -171000)
```

Comment:
Is this correct?

`irods error: -171000 NOT_A_DATA_OBJECT`

Blindly, I would have expected `-317000 USER_INPUT_PATH_ERR` or `-808000 CAT_NO_ROWS_FOUND`...

Reply:
I tried a few different invalid paths and it always gives back -171000.

Reply:
fair enough - please add a comment here for the `NOT_A_DATA_OBJECT` part.

Reply:
The HTTP API does perform input verification for some of the operations (e.g. is the logical path pointing to a collection). This may be one of those situations.

Reply:
Opened issue #18